### ISSUE TITLE
Can Git for Windows be configured to use msys2/mingw binaries directly?

### ISSUE THREAD

**Issue by vonHabsi** [git-for-windows/git#904]
I installed msys2/mingw64 and I also have Git for Windows and it seems that some path related issues came up. Aspects like the use of ssh keys, tmux etc also seem to be much easier make working with Git easier under msys2/mingw.

If I am not interested in the GUI will it be better to use the command line in mingw64/msys2 or use Git for Windows? Even if I resort to using some of the GUI tools occasionally,like reviewing branches etc are there bound to be some conflicts?

Are there some non-GUI customisations made for Git for Windows that make it different or might cause conflict with msys2 environment? It is possible to install Git for Windows and make its shell related settings and non GUI binaries point to those of the msys2 environment without any unwanted effects?

So in a nutshell is Git for Windows a barebones set of msys2 configured to work out of the C:\Program Files directory?

**Comment by nalla** [comment]
> So in a nutshell is Git for Windows a barebones set of msys2 configured to work out of the C:\Program Files directory?

Exactly.

If you want the power of msys2 combined with the power of _Git for Windows_ you should use the _Git for Windows SDK_. Which itself is just a fork of msys2 with some extra packages like the mingw64 git package.

**Comment by vonHabsi** [comment]
I think I didn't express myself properly. I don't want to compile Git for Windows as such. What I want to know is that the non GUI parts can be configured to run using the same binaries in the msys2 installation.
eg, can it use the ssh, vim, git etc in msys2, and can its environment be set to match msys2 so that when it comes to .bashrc, .bash_profile, .vimrc etc it will automatically use those in msys2, and only use the GUI parts of installation. When Git for Windows installed it makes changes to the PATH variable, and configures some shell extensions etc. Can its install time configuration be made to use more of command line executables in msys2?

**Comment by nalla** [comment]
I think I didn't express myself properly. If you use the Git for Windows SDK you will have fully-functional msys2 environment with a few additional packages available in pacman. One of those packages is https://github.com/git-for-windows/MINGW-packages/tree/master/mingw-w64-git which will be installed by default when installing the Git for Windows SDK.

The resulting Git for Windows SDK installation is nothing more than an msys2 installation resulting in exactly the behavior you want.

**Comment by dscho** [comment]
The path related issues most likely stem from the fact that Git for Windows uses a modified MSYS2 runtime, specifically to address the needs of Git.
